core/src/main/java/org/infinispan/expiration/ExpirationManager.java

```diff
@@ -35,11 +35,32 @@
    boolean isEnabled();
 
    /**
-    * This should be invoked passing in an entry that is now expired.  Note this method assumes to not have the
-    * lock for this key and will attempt to acquire it.
+    * This should be invoked passing in an entry that is now expired.  This method assumes the lock is not acquire
+    * and may attempt to lock this key to preserve atomicity.
```

Comment:
I think it would be better to say it doesn't make any assumptions about the key being locked, because it should work if the current tx has the lock.